test/test_test.lua

```diff
@@ -0,0 +1,36 @@
+local expect_raw = reqscript('test/main').expect
```

Comment:
> rename test.lua to test_test.lua since I plan to rename main.lua to test.lua in a future PR

I think we should reconsider where the test runner lives in the future, in that case. I didn't really think through where the runner should live, but it shouldn't be in the same namespace as the tests themselves. Essentially, my proposal is to:
* move test/main.lua to somewhere else in the DFHack repo and install it to hack/scripts if BUILD_TESTS is enabled. (Now that I think about it, it probably lives at `test/main` because I couldn't think of another place to put it in the DFHack repo, but maybe the `ci` folder could work.)
* don't add the `test` folder to the script path, since none of the test cases themselves are intended to be run directly (and they wouldn't work if they were). This would allow tests to have the same names as other scripts.

This is something I can handle if it's not too disruptive to your test work. Just wanted to make sure our plans were aligned - feel free to suggest something else.

Reply:
that sounds good to me. Changes to the test runner don't conflict with anything I have in the pipeline. I'm just writing unit tests ATM.

Now that we're going to install the test runner directly into the `scripts/` directory, I do suggest still renaming `main.lua` to `test.lua`. Running tests with `test/main` made sense. Running tests with just `main` isn't quite as intuitive.

Reply:
Oh, definitely - forgot to mention it, but I was thinking the same thing.